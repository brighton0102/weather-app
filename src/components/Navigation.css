.nav-container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: rgb(75, 103, 179);
  padding: 20px;
}

.navhead {
  padding-left: 38px;
  font-size: 20px;
}

.nav-icons {
  display: flex;
  gap: 20px;
  padding-right: 10px;
}
