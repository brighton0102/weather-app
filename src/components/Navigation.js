import React from 'react';
import { IoIosArrowBack, IoMdMic, IoMdSettings } from 'react-icons/io';
import { Link } from 'react-router-dom';
import './Navigation.css'

function Navigation() {
  return (
    <nav>
      <div className="nav-container">
        <Link to="/">
          <IoIosArrowBack size={30} color="#007bff" />
        </Link>
        <p className="navhead">CityWeather</p>
        <div className="nav-icons">
          <IoMdMic size={24} color="green" />
          <IoMdSettings size={24} color="red" />
        </div>
      </div>
    </nav>
  );
}

export default Navigation;
