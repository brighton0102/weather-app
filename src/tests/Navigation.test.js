import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';

test('Navigation component renders correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});
