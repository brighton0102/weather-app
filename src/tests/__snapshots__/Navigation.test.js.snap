// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<div>
  <nav>
    <div
      class="nav-container"
    >
      <a
        href="/"
      >
        <svg
          color="#007bff"
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(0, 123, 255);"
          viewBox="0 0 512 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </a>
      <p
        class="navhead"
      >
        CityWeather
      </p>
      <div
        class="nav-icons"
      >
        <svg
          color="green"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          style="color: green;"
          viewBox="0 0 512 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 320c37.712 0 68.571-30.924 68.571-68.714V100.714C324.571 62.924 293.712 32 256 32s-68.571 30.924-68.571 68.714v150.572c0 37.79 30.859 68.714 68.571 68.714zm121.139-75.452c0 68.714-58.282 116.815-121.139 116.815s-121.139-48.102-121.139-116.815H96c0 77.873 61.719 143.153 137.144 153.465V480h45.713v-81.987C354.281 386.561 416 322.421 416 244.548h-38.861z"
          />
        </svg>
        <svg
          color="red"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          style="color: red;"
          viewBox="0 0 512 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.967 276.8c1.06-6.235 1.06-13.518 1.06-20.8s-1.06-13.518-1.06-20.8l44.667-34.318c4.26-3.118 5.319-8.317 2.13-13.518L418.215 115.6c-2.129-4.164-8.507-6.235-12.767-4.164l-53.186 20.801c-10.638-8.318-23.394-15.601-36.16-20.801l-7.448-55.117c-1.06-4.154-5.319-8.318-10.638-8.318h-85.098c-5.318 0-9.577 4.164-10.637 8.318l-8.508 55.117c-12.767 5.2-24.464 12.482-36.171 20.801l-53.186-20.801c-5.319-2.071-10.638 0-12.767 4.164L49.1 187.365c-2.119 4.153-1.061 10.399 2.129 13.518L96.97 235.2c0 7.282-1.06 13.518-1.06 20.8s1.06 13.518 1.06 20.8l-44.668 34.318c-4.26 3.118-5.318 8.317-2.13 13.518L92.721 396.4c2.13 4.164 8.508 6.235 12.767 4.164l53.187-20.801c10.637 8.318 23.394 15.601 36.16 20.801l8.508 55.117c1.069 5.2 5.318 8.318 10.637 8.318h85.098c5.319 0 9.578-4.164 10.638-8.318l8.518-55.117c12.757-5.2 24.464-12.482 36.16-20.801l53.187 20.801c5.318 2.071 10.637 0 12.767-4.164l42.549-71.765c2.129-4.153 1.06-10.399-2.13-13.518l-46.8-34.317zm-158.499 52c-41.489 0-74.46-32.235-74.46-72.8s32.971-72.8 74.46-72.8 74.461 32.235 74.461 72.8-32.972 72.8-74.461 72.8z"
          />
        </svg>
      </div>
    </div>
  </nav>
</div>
`;
